core/src/test/java/com/linecorp/armeria/common/ResponsePeekTest.java

```diff
@@ -48,25 +49,125 @@ void peekHeaders() {
+                .mapTrailers(trailers -> {
+                    invoked.set(true);
+                    return trailers;
+                });
+
+        final AggregatedHttpResponse aggregated1 = transformed1.aggregate().join();
+        assertThat(invoked).isFalse();
+        assertThat(peekCount).hasValue(0);
+        assertThat(aggregated1.headers().status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregated1.contentUtf8()).isEqualTo("foo");
+        assertThat(aggregated1.trailers().isEmpty()).isTrue();
+
+        final HttpResponse withTrailers = HttpResponse.of(headers,
+                                                          HttpData.ofUtf8("foo"),
+                                                          HttpHeaders.of("status", "13"));
+
+        final HttpResponse transformed2 = withTrailers
+                .peekTrailers(trailers -> {
+                    assertThat(trailers.get("ok")).isNull();
+                    assertThat(trailers.get("error")).isNull();
+                })
+                .mapTrailers(trailers -> {
+                    invoked.set(true);
+                    if ("13".equals(trailers.get("status"))) {
+                        return trailers.toBuilder().add("error", "INTERNAL").build();
+                    } else {
+                        return trailers.toBuilder().add("ok", "true").build();
+                    }
+                })
+                .peekTrailers(trailers -> {
+                    assertThat(trailers.get("ok")).isNull();
+                    assertThat(trailers.get("error")).isEqualTo("INTERNAL");
+                })
+                .peekTrailers(trailers -> peekCount.incrementAndGet());
+
+        final AggregatedHttpResponse aggregated2 = transformed2.aggregate().join();
+        assertThat(invoked).isTrue();
+        assertThat(peekCount).hasValue(1);
+        assertThat(aggregated2.headers().get("ok")).isNull();
+        assertThat(aggregated2.headers().get("error")).isNull();
+        assertThat(aggregated2.contentUtf8()).isEqualTo("foo");
+        assertThat(aggregated2.trailers().size()).isEqualTo(2);
+        assertThat(aggregated2.trailers().get("status")).isEqualTo("13");
+        assertThat(aggregated2.trailers().get("error")).isEqualTo("INTERNAL");
+        assertThat(aggregated2.trailers().get("ok")).isNull();
+    }
+
+    @Test
+    void peekChain() {
         final HttpResponse res = HttpResponse.of(ResponseHeaders.of(HttpStatus.OK),
                                                  HttpData.ofUtf8("foo"),
                                                  HttpHeaders.of("status", "0"));
 
         final AggregatedHttpResponse aggregated =
-                res.mapHeaders(headers -> headers.toBuilder().add("header1", "1").build())
+                res.peekHeaders(headers -> {
+                    assertThat(headers.get("header1")).isNull();
+                    assertThat(headers.get("header2")).isNull();
+                })
```

Comment:
nit; the indentation seems a little off here

```suggestion
                        assertThat(headers.get("header1")).isNull();
                        assertThat(headers.get("header2")).isNull();
                    })
```